src/components/ToggleFavorite/ToggleFavorite.js

```diff
@@ -1,25 +1,15 @@
 // ToggleFavorite //
 import './ToggleFavorite.css'
 
-const ToggleFavorite = ({quote, setQuote, favorites, setFavorites}) => {
-
-  const handleFavorite = () => { 
-    if (quote.isFavorite) {
-
-      setQuote({...quote, isFavorite: false})
-    } else {
-      setFavorites([...favorites, quote])
-      setQuote({...quote, isFavorite: true})    
-    }
-  }
+const ToggleFavorite = ({quote, handleFavorite}) => {
 
   if (quote.isFavorite) {
     return (
-      <button className="card__favorite" onClick={handleFavorite}>❤️</button>
+      <button className="card__favorite" id={quote._id} onClick={handleFavorite}>❤️</button>
     ) 
   } else { 
     return (
-      <button className="card__favorite" onClick={handleFavorite}>🤍</button>
+      <button className="card__favorite" id={quote._id} onClick={handleFavorite}>🤍</button>
     )
   }
```

Comment:
This would be an interesting use case for compound components. This is a small app, but it wouldn't hurt to get some practice with them. [Here's](https://kentcdodds.com/blog/compound-components-with-react-hooks) a good lesson on them, they are very important to React app development.